File: HwaSrsApi/Models/CardType.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace HwaSrsApi.Models
{
    public class CardType
    {
        public int Id { get; set; }
        [MaxLength(64)]
        [Required]
        public string Name { get; set; }
        public bool ReadOnce { get; set; }
        [Required]
        public string Design { get; set; }

        [JsonIgnore]
        public virtual List<Card> Cards { get; set; } = new List<Card>();
        public virtual List<Field> Fields { get; set; } = new List<Field>();
    }
}

File: src/app/types/card-type.ts
interface CardType {
    id: number;
    name: string;
    readOnce: boolean;
    design: string;
    fields: Field[];
}

interface Field {
    id: number;
    name: string;
    cardTypeId: CardType['id'];
}

interface NewCardType extends Omit<CardType, 'id' | 'fields'> {
    fields: NewField[];
}

type NewField = Omit<Field, 'id' | 'cardTypeId'>;

export {
    CardType,
    Field,
    NewCardType,
    NewField,
};
